thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/HoltWintersDetector.java

```diff
@@ -152,24 +153,30 @@ public TimeSeries computePredictedTimeSeries(MetricSlice slice) {
     return TimeSeries.fromDataFrame(resultDF);
   }
 
-  private DateTime getTrainingStartTime(Interval window) {
+  private DateTime getTrainingStartTime(DateTime windowStart) {
     DateTime trainStart;
     if (isMultiDayGranularity()) {
-      trainStart = window.getStart().minusDays(timeGranularity.getSize() * LOOKBACK);
+      trainStart = windowStart.minusDays(timeGranularity.getSize() * LOOKBACK);
     } else if (this.monitoringGranularity.endsWith(TimeGranularity.MONTHS)) {
-      trainStart = window.getStart().minusMonths(LOOKBACK);
+      trainStart = windowStart.minusMonths(LOOKBACK);
     } else if (this.monitoringGranularity.endsWith(TimeGranularity.WEEKS)) {
-      trainStart = window.getStart().withTimeAtStartOfDay().withDayOfWeek(weekStart.getValue()).minusWeeks(LOOKBACK);
+      trainStart = windowStart.minusWeeks(LOOKBACK);
     } else {
-      trainStart = window.getStart().minusDays(LOOKBACK);
+      trainStart = windowStart.minusDays(LOOKBACK);
     }
     return trainStart;
   }
 
   @Override
   public DetectionResult runDetection(Interval window, String metricUrn) {
     MetricEntity metricEntity = MetricEntity.fromURN(metricUrn);
-    DateTime trainStart = getTrainingStartTime(window);
+    DateTime windowStart = window.getStart();
+    // align start day to the user specified week start
+    if (Objects.nonNull(this.weekStart)) {
```

Comment:
If user doesn't define a weekStart we should use a default value. Isn't it?

Reply:
Yes. The `weekStart` will have a default value if the granularity is set to weekly.

Reply:
Got it. Thanks.